### PR TITLE
Skip cuTENSOR/cuSPARSELt installation for unsupported CUDA version

### PR DESCRIPTION
**Pull request by asi1024**

### CODE REVIEW

**Comment by leofang** [comment]
#5136 may be relevant.

File: cupy_setup_build.py [hunk 1]
```diff
@@ -526,6 +527,14 @@ def preconfigure_modules(compiler, settings):
             cmd = 'nvcc' if not use_hip else 'hipcc'
             errmsg = ['{} command could not be found in PATH.'.format(cmd),
                       'Check your PATH environment variable.']
+        elif ('required_cuda_version' in module
```
Comment (kmaehashi) [inline cupy_setup_build.py:530]: This needs to be checked first (before `not check_library()` condition) to skip building cuTENSOR/cuSPARSELt support for unsupported CUDA versions.

File: cupy_setup_build.py [hunk 2]
```diff
@@ -526,6 +527,14 @@ def preconfigure_modules(compiler, settings):
...
+              and not module['check_cuda_version'](build.get_cuda_version())):
+            cuda_version = build.get_cuda_version()
+            cuda_major = cuda_version // 1000
+            cuda_major_minor = f'{cuda_major}.{(cuda_version // 10) % 100}'
+            installed = True
+            errmsg = [
+                f'The library is not supported for CUDA {cuda_major_minor}']
```
Comment (kmaehashi) [inline cupy_setup_build.py:537]:
```suggestion
                f'Not supported in CUDA {cuda_major_minor}']
```

**Review by kmaehashi — CHANGES_REQUESTED** [review]

**Comment by kmaehashi** [comment]
I think this doesn't make much sense after dropping CUDA 10.2... maybe it's ok to close? @asi1024 :bow:

**Pull request closed without merging**